### PR TITLE
OSD-24614 : Update osdctl to add justification arguement for infranode resize

### PR DESCRIPTION
**Pull request by MitaliBhalla**
Our current infra node resize [SL](https://github.com/openshift/managed-notifications/blob/master/osd/infranode_resized.json) does mention about the justification of the resize, . However, in our [SOP] (https://github.com/openshift/ops-sop/blob/master/v4/alerts/InfraNodesNeedResizingSRE.md) , the command mentioned doesn't mention about the justification arguement as osdctl refers to other SL that doesn't have the justification arguement. https://github.com/openshift/osdctl/blob/master/cmd/cluster/resize/infra_node.go#L35C43-L35C147
This PR incorporates for :
1. osdctl to point to the SL that has justification arguement 
2. Add changes accordingly to incorporate the justification arguement

### CODE REVIEW

File: cmd/cluster/resize/infra_node.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ import (
 const (
 	twentyMinuteTimeout                   = 20 * time.Minute
 	twentySecondIncrement                 = 20 * time.Second
-	resizedInfraNodeServiceLogTemplate    = "https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/infranode_resized_auto.json"
+	resizedInfraNodeServiceLogTemplate    = "https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/infranode_resized.json"
 	GCPresizedInfraNodeServiceLogTemplate = "https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/gcp/GCP_infranode_resized_auto.json"
```
Comment (devppratik) [inline cmd/cluster/resize/infra_node.go:36]: /hold
Waiting until the service log are properly updated

File: cmd/cluster/resize/infra_node.go [hunk 2]
```diff
@@ -536,18 +542,18 @@ func getInstanceType(mp *hivev1.MachinePool) (string, error) {
...
 
 // Adding change in serviceLog as per the cloud provider.
-func generateServiceLog(mp *hivev1.MachinePool, instanceType, clusterId string) servicelog.PostCmdOptions {
+func generateServiceLog(mp *hivev1.MachinePool, instanceType, justification, clusterId string) servicelog.PostCmdOptions {
 	if mp.Spec.Platform.AWS != nil {
 		return servicelog.PostCmdOptions{
 			Template:       resizedInfraNodeServiceLogTemplate,
 			ClusterId:      clusterId,
-			TemplateParams: []string{fmt.Sprintf("INSTANCE_TYPE=%s", instanceType)},
+			TemplateParams: []string{fmt.Sprintf("INSTANCE_TYPE=%s", instanceType),fmt.Sprintf("JUSTIFICATION=%s", justification)},
```
Comment (devppratik) [inline cmd/cluster/resize/infra_node.go:550]:
```suggestion
			TemplateParams: []string{fmt.Sprintf("INSTANCE_TYPE=%s", instanceType),fmt.Sprintf("JUSTIFICATION=%s", justification)}
```

File: cmd/cluster/resize/infra_node.go [hunk 3]
```diff
@@ -536,18 +542,18 @@ func getInstanceType(mp *hivev1.MachinePool) (string, error) {
...
 		}
 	} else if mp.Spec.Platform.GCP != nil {
 		return servicelog.PostCmdOptions{
 			Template:       GCPresizedInfraNodeServiceLogTemplate,
 			ClusterId:      clusterId,
-			TemplateParams: []string{fmt.Sprintf("INSTANCE_TYPE=%s", instanceType)},
+			TemplateParams: []string{fmt.Sprintf("INSTANCE_TYPE=%s", instanceType),fmt.Sprintf("JUSTIFICATION=%s", justification)},
```
Comment (devppratik) [inline cmd/cluster/resize/infra_node.go:556]:
```suggestion
			TemplateParams: []string{fmt.Sprintf("INSTANCE_TYPE=%s", instanceType),fmt.Sprintf("JUSTIFICATION=%s", justification)}
```

**Review by devppratik — COMMENTED** [review]
Fixing the lint failures. The trailing commas seems to be causing the issue

File: cmd/cluster/resize/infra_node.go (hunk 1 above)
Comment (MitaliBhalla) [reply cmd/cluster/resize/infra_node.go:36]: Done via https://github.com/openshift/managed-notifications/pull/327

**Pull request merged**